agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/profiler/ProfilingInitializer.java

```diff
@@ -127,17 +132,40 @@ private synchronized void performInit() {
             userAgent);
 
     // Monitor service remains alive permanently due to scheduling an periodic config pull
-    ConfigPollingInit.startPollingForConfigUpdates(
-        serviceProfilerExecutorService,
-        serviceProfilerClient,
-        this::applyConfiguration,
-        configuration.preview.profiler.configPollPeriodSeconds);
+    startPollingForConfigUpdates();
+  }
+
+  private void startPollingForConfigUpdates() {
+    ConfigService configService = new ConfigService(serviceProfilerClient);
+    serviceProfilerExecutorService.scheduleAtFixedRate(
+        () -> {
+          try {
+            configService
+                .pullSettings()
+                .subscribe(
+                    this::applyConfiguration,
+                    e -> {
+                      if (currentlyEnabled.get()) {
+                        logger.error("Error pulling service profiler settings", e);
+                      }
```

Comment:
else `logger.debug` may be helpful for debugging customers who aren't able to make the initial connection for some reason (but say they have configured profiling in portal)